library/ecdh.c

```diff
@@ -191,19 +191,28 @@ int mbedtls_ecdh_make_params( mbedtls_ecdh_context *ctx, size_t *olen,
 {
     int ret;
     size_t grp_len, pt_len;
+#if !defined(MBEDTLS_ECDH_GEN_PUBLIC_ALT)
     mbedtls_ecp_restart_ctx *rs_ctx = NULL;
+#endif
 
     if( ctx == NULL || ctx->grp.pbits == 0 )
         return( MBEDTLS_ERR_ECP_BAD_INPUT_DATA );
 
-#if defined(MBEDTLS_ECP_RESTARTABLE)
+#if defined(MBEDTLS_ECP_RESTARTABLE) && !defined(MBEDTLS_ECDH_GEN_PUBLIC_ALT)
     if( ctx->restart_enabled )
         rs_ctx = &ctx->rs;
 #endif
 
+
+#if defined(MBEDTLS_ECDH_GEN_PUBLIC_ALT)
+    if( ( ret = mbedtls_ecdh_gen_public( &ctx->grp, &ctx->d, &ctx->Q,
+                                         f_rng, p_rng ) ) != 0 )
+        return( ret );
+#else
     if( ( ret = ecdh_gen_public_restartable( &ctx->grp, &ctx->d, &ctx->Q,
                                              f_rng, p_rng, rs_ctx ) ) != 0 )
         return( ret );
+#endif /* MBEDTLS_ECDH_GEN_PUBLIC_ALT */
```

Comment:
```suggestion
#endif /* !MBEDTLS_ECDH_GEN_PUBLIC_ALT */
```

Reply:
but the check is for [`#if defined(MBEDTLS_ECDH_GEN_PUBLIC_ALT)`](https://github.com/ARMmbed/mbedtls/pull/2165/files#diff-97f71deba621674b90985642f0884ebdR207)

Reply:
There's an `#else`

Reply:
yes, but we usually comment what `#if defined` this closes.
I'll change it

Reply:
Huh, seems that we do. Weird, I've never seen this convention before. Please don't change it, it's better to be consistent inside the library.